src/347-Top-K-Frequent-Elements/347._Top_K_Frequent_Elements.cpp

```diff
@@ -0,0 +1,49 @@
+class Solution {
+public:
+    //use vector
+    vector<int> topKFrequent_approach1(vector<int>& nums, int k) {
+        vector<int> res;
+        vector<pair<int, int>> pairCur;
+        vector<int>::iterator it;
+        sort(nums.begin(), nums.end());
+        for (it = nums.begin(); it != nums.end(); it++) {
+            int count = 1;
+            while ((it + 1) != nums.end() && *(it+1) == *it) {
+                count++;
+                it++;
+            }
+            pairCur.push_back(pair<int, int>(*it, count));
+        }
+        auto cmp = [](const pair<int,int>& left, const pair<int, int>& right) {
+            return left.second > right.second;
+        };
+        
+        sort(pairCur.begin(), pairCur.end(), cmp);
```

Comment:
Since cmp is too short, we could use it directly like this:
```c++
sort(pairCur.begin(), pairCur.end(),  [](const pair<int,int>& left, const pair<int, int>& right) {
            return left.second > right.second;
        };);
```